### PR TITLE
Fixes for handling `SCYLLA_EXT_OPTS` correctly

### PR DESCRIPTION
**Pull request by eliransin**
This mini-series implements two fixes for handling `SCYLLA_EXT_OPTS`:
1. It adds support for `-c` and `-m` options which are abbreviations for the `--smp` and `--memory` options and handles them correctly.
2. It adds support for multiple occurrences of  `--experimental-features` with correct handling.

The individual issues:
#539 and #540 describes the specific problems.

### CODE REVIEW

**Comment by tchaikov** [comment]
I will review this change once I am back from my PTO.

**Comment by eliransin** [comment]
@tchaikov ping

File: ccmlib/scylla_node.py [hunk 1]
```diff
@@ -570,11 +570,12 @@ def process_opts(opts):
                         opts_i += 1
                     val = ' '.join(vals)
                 if key not in ext_args and not key.startswith("--scylla-manager"):
-                    ext_args[key] = val
+                    ext_args[args_translation_map.get(key, key)] = val
```
Comment (tchaikov) [inline ccmlib/scylla_node.py]: i'd suggest normalize `key` even before the `if key not in ext_args and not key.startswith("--scylla-manager")` check, and do this unconditially. like:
```py
key = args_translation_map.get(key, key)
if key not in ext_args and not key.startswith("--scylla-manager"):
    ext_args[key] = val
```

for better readaiblity.

File: ccmlib/scylla_node.py [hunk 2]
```diff
@@ -549,8 +549,15 @@ def start(self, join_ring=True, no_wait=False, verbose=False,
 
         MB = 1024 * 1024
 
-        def process_opts(opts, args_translation_map = {}):
+        def process_opts(opts, args_translation_map = {}, allowed_duplicates_set = set()):
             ext_args = OrderedDict()
+            def should_add_option(key):
```
Comment (tchaikov) [inline ccmlib/scylla_node.py:554]: the nested function is more readable, but it also renders this method more complicated, how about this?

```py
if key.startswith("--scylla-manager"):
    continue

if key in allowed_duplicates_set:
    ext_args.get(key, []).append(val)
elif key not in ext_args:
    ext_args[key] = val
```

for following reasons:

1. the combinations of conditions is getting convoluted in this function, better off simplifying them
2. `dict.set_default()` helps to consolidate the logic, but it hurts the readability.
3. we should not use list for all values of options. the boxing-unboxing hurts the readability.

**Comment by tchaikov** [comment]
@eliransin hi Eliran, sorry for the latency. left a couple comments.

**Comment by fruch** [comment]
closing in favor of #558

**Pull request closed without merging**